xbmc/addons/include/kodi_vfs_types.h

```diff
@@ -0,0 +1,32 @@
+/*
+ *      Copyright (C) 2015 Team Kodi
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Kodi; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
+#include <stdint.h>
+
+extern "C"
+{
+  struct VFSDirEntry
+  {
+    char* label;             //!< item label
+    char* path;              //!< item path
+    bool folder;             //!< Item is a folder
+    uint64_t size;           //!< Size of file represented by item
+  };
+}
```

Comment:
Ok for you to this on end in related library header e.g.
https://github.com/AlwinEsch/kodi/blob/binary-lib-rework-2/addons/library.kodi.addon/addon.api2/definations.h
or
https://github.com/AlwinEsch/kodi/blob/binary-lib-rework-2/addons/library.kodi.addon/addon.api2/Directory.h
?